src/main/java/com/aws/greengrass/clientdevices/auth/iot/Certificate.java

```diff
@@ -46,7 +46,11 @@ public enum Status {
     private Instant statusLastUpdated;
 
 
-    Certificate(String certificateId) {
+    /**
+     * Builds a new Certificate instance given its id generated from PEM file.
+     * @param certificateId - a certificateId
+     */
+    public Certificate(String certificateId) {
```

Comment:
I actually had this package private as a hint that nothing outside of the package should instantiate a Certificate directly. It looks like you just had to make this public for tests. Maybe we can move that test into the same package?

I probably should have commented my motivation for doing that to make it clear.

Reply:
It was not for the tests but for the BackgroundRefresh and avoid getting the PEM first but I can rework this